### PR TITLE
Fix hydration for hooks

### PR DESCRIPTION
**Pull request by mocca102**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -88,6 +89,12 @@ export type AnswerInputState = {
   };
 };
 
+export interface QuizSessionStorageState {
```
Comment (mocca102) [inline src/types.ts:92]: Separate session storage state

File: src/hooks/useQuizState/useSessionStorageState.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react';
```
Comment (mocca102) [inline src/hooks/useQuizState/useSessionStorageState.ts:1]: Hook for handling session storage state

File: src/hooks/useQuizEvents/useHydrateQuizLocalState.ts [hunk 3]
```diff
@@ -4,18 +4,18 @@ import { QuizEventsReturn } from '../../types';
 import { getStateFromSessionStorage } from '../../utils';
 
 const useHydrateQuizLocalState = (
-  quizStateKey: string,
+  quizSessionStorageStateKey: string,
```
Comment (mocca102) [inline src/hooks/useQuizEvents/useHydrateQuizLocalState.ts:7]: Rename for clarity

File: src/hooks/useQuiz.ts [hunk 4]
```diff
@@ -38,11 +45,11 @@ const useQuiz: UseQuiz = (quizOptions) => {
         requestState: quizApiState.quizRequestState,
         versionId: quizLocalState.quizVersionId,
         sessionId: quizLocalState.quizSessionId,
-        skipToResults,
```
Comment (mocca102) [inline src/hooks/useQuiz.ts:41]: moved to quizSessionStorageState

File: src/hooks/useQuiz.ts [hunk 5]
```diff
@@ -22,11 +23,17 @@ const useQuiz: UseQuiz = (quizOptions) => {
...
+  const { skipToResults } = quizSessionStorageState;
   const propGetters = usePropsGetters(quizEvents, quizApiState, quizLocalState);
 
   const primaryColorStyles = usePrimaryColorStyles(primaryColor);
 
+  useEffect(() => {
+    if (skipToResults) quizEvents.hydrateQuiz();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
```
Comment (mocca102) [inline src/hooks/useQuiz.ts:36]: Always hydrate results if skipToResults is true

Comment (mocca102) [reply src/hooks/useQuiz.ts:36]: skipToResults is true when

- Hydration is enabled in the config object
- The quiz is completed
- And showSessionModalOnResults is false

File: src/components/CioQuiz/index.tsx [hunk 6]
```diff
@@ -29,24 +30,26 @@ export default function CioQuiz(props: IQuizProps) {
...
-          hasSessionStorageState() &&
-          !state.quiz.skipToResults
+        sessionStateOptions?.showSessionModal && hasSessionStorageState() && !skipToResults
       );
     } else {
-      setShowSessionPrompt(hasSessionStorageState() && !state.quiz.skipToResults);
+      setShowSessionPrompt(hasSessionStorageState() && !skipToResults);
     }
-
-    if (state.quiz.skipToResults) hydrateQuiz();
```
Comment (mocca102) [inline src/components/CioQuiz/index.tsx:49]: This is handled by hooks now

File: src/components/CioQuiz/index.tsx [hunk 7]
```diff
@@ -29,24 +30,26 @@ export default function CioQuiz(props: IQuizProps) {
     getResetQuizButtonProps,
     getSelectInputProps,
     primaryColorStyles,
-  } = useQuiz(props);
+  } = useQuiz({
+    ...props,
+    enableHydration: enableHydration === undefined ? true : enableHydration,
```
Comment (mocca102) [inline src/components/CioQuiz/index.tsx:35]: If enableHydration isn't passed in the config object default true. if it's passed in the config object use the passed value instead

Comment (esezen) [reply src/components/CioQuiz/index.tsx:35]: @mocca102 could this logic be moved to useQuiz? What's the benefit of having it here instead of in the hook?

**Review by esezen — APPROVED** [review]
LGTM!

**Pull request merged**